sdk/sdk-skeleton.go

```diff
@@ -0,0 +1,77 @@
+type Asset struct{}
+
+//// Methods directly on the SDK
+
+func NewChannel() {
+
+}
+
+func NewChannelListenerService() {
+
+}
+
+//// Channel State Machine
+
+type ChannelStateMachine struct {
+	Status                 string // waiting | active | closing | closed
+	Asset                  Asset
+	InitiatorEscrowAccount string
+	ResponderEscrowAccount string
+	InitiatorClaimAmount   int
+	ResponderClaimAmount   int
+	IsInitiator            bool
+	MyAccount              string
```

Comment:
Do we need `MyClaimAmount` and `OtherClaimAmount`. Aren't those describing `InitiatorClaimAmount` and `ResponderClaimAmount`?
cc @nikhilsaraf

Reply:
Regarding the amounts I think there's two different things we need to know about:
- A cache of the last known balance of the escrow accounts. This value may go up without us knowing, but it will only go down with coordination. These amounts need to go down immediately for a signed withdrawal, but the balance of the escrow account won't actually go down immediately, and may never go down if the withdrawal fails, so there's some complexity we need to figure out.
- The amount that each participant owes the other participant. Payments should move these values. We could instead store the total amount paid in both directions and calculate the amount owing as it is needed, but that approach would be riskier for overflows.

Reply:
`MyClaimAmount` and `OtherClaimAmount` is intended to only be a convenience method. i.e. the SDK could be either the Initiator or Responder. The `is_initiator` bool allows them to check and pick the right account, so the `MyClaimAmount` just makes this easier -- this should probably be a function instead of a variable on the struct

Reply:
moved to helper methods:
https://github.com/stellar/experimental-payment-channels/pull/25/commits/5bd4f54a1d3e26e434e2f21521191fde56c108d9